mh/constants.py

```diff
@@ -164,13 +166,43 @@ def make_binary_npc_greeters(is_jap=False):
     US_OFFSET = 0x180
     JP_OFFSET = 0x100
     offset = JP_OFFSET if is_jap else US_OFFSET
+
+    day_in_cycle = int(current_server_time()//SECONDS_PER_DAY) % JHEN_EVENT_OFFSET
+    is_jhen_active = day_in_cycle >= JHEN_START and day_in_cycle < JHEN_END
+
+    if is_jhen_active:
+        tool_shop_str = "Half-off sale!"
+        material_shop_str = "Half-off sale!"
+        event_quests_str = "The Festival of Fear\n"
+    else:
+        tool_shop_str = "Prices are normal."
+        material_shop_str = "Prices are normal."
+        event_quests_str = ""
+
+    event_quests_str += QUEST_EVENT_JUMP_FOUR_JAGGI[0:40].decode('ascii').rstrip('\x00') + '\n'
+    event_quests_str += QUEST_EVENT_BLOOD_SPORT[0:40].decode('ascii').rstrip('\x00') + '\n'
+    event_quests_str += QUEST_EVENT_MERCY_MISSION[0:40].decode('ascii').rstrip('\x00') + '\n'
+    event_quests_str += QUEST_EVENT_THE_PHANTOM_URAGAAN[0:40].decode('ascii').rstrip('\x00') + '\n'
+    event_quests_str += QUEST_EVENT_WORLD_EATER[0:40].decode('ascii').rstrip('\x00') + '\n'
+    event_quests_str += QUEST_EVENT_WHERE_GODS_FEAR_TO_TREAD[0:40].decode('ascii').rstrip('\x00') + '\n'
+    event_quests_str += QUEST_EVENT_GREEN_EGGS[0:40].decode('ascii').rstrip('\x00')
+
+    guildmaster_str = "To all hunters:\n"
+    guildmaster_str += "Loc Lac is in \"beta,\"\n"
+    guildmaster_str += "whatever that means. Also,\n"
+    guildmaster_str += "the Arena is currently under\n"
+    guildmaster_str += "construction. And now, a:\n"
+    guildmaster_str += "haiku: \"Patient warriors /\n"    
+    guildmaster_str += "returning home at long last /\n"
+    guildmaster_str += "now join hands to hunt!\""
+
     data = b""
-    data += pad(b"Plaza Tool Shop\n\nNot supported yet.", offset)
-    data += pad(b"Material shop unavailable\nyet.", offset)
-    data += pad(b"Trading post closed at\nthe moment.", offset)
-    data += pad(b"No event quests.", offset)
+    data += pad(tool_shop_str.encode(), offset)
+    data += pad(material_shop_str.encode(), offset)
+    data += pad(b"The Trading Post is open!", offset)
+    data += pad(event_quests_str.encode(), offset)
     data += pad(b"No arena quests.", offset)
-    data += pad(b"To all hunters:\n\nThis is a test server.", offset)
+    data += pad(guildmaster_str.encode(), offset)
```

Comment:
Please make these additions by creating another PR.

Reply:
The title of this PR has been changed to be more accurate to what it's supposed to do